src/buildercore/bootstrap.py

```diff
@@ -111,6 +111,8 @@ def stack_resources(stackname):
 
 def ec2_instance_data(stackname):
     "returns the ec2 instance data from the first ec2 instance the stack has"
+    if stackname is None:
```

Comment:
I prefer assertions to  raising RuntimeErrors. so, make these `assert stackname, "no stackname provided"`.

the absence of a stackname is programmer error and not something outside of our control.